### ISSUE TITLE
Duplicate card titles seem to cause trouble

### ISSUE THREAD

**Issue by jhheider** [websitesfortrello/trelew#3]
`Warning: command named "card batteries" was registered more than once.
If you intend to override a command, you should explicitly remove the first command with command.remove().
Warning: command named "card potion-bottles" was registered more than once.
If you intend to override a command, you should explicitly remove the first command with command.remove().
Warning: command named "card sam-s-order" was registered more than once.
If you intend to override a command, you should explicitly remove the first command with command.remove().
Warning: command named "card gts-and-alliance-costs" was registered more than once.
If you intend to override a command, you should explicitly remove the first command with command.remove().
Warning: command named "card sam-s-order" was registered more than once.
If you intend to override a command, you should explicitly remove the first command with command.remove().`

Very cool project, and, based on the addressing scheme, this might be an annoying problem to solve. If I get a chance, I will dig into the source and see if I can generate a fix.

**Comment by fiatjaf** [comment]
Yes, that's an important issue.

I tried to address it myself -- right on the function that generates slugs for being later used as commands -- with the following code: https://github.com/websitesfortrello/trelew/blob/4f34b2fedb62676041a5c71300ff93f4edbc20c3/helpers.js#L73-L94

But sadly it doesn't work as expected. Also, the expected behavior (replacing the last characters of the slug with numbers) wasn't the best.
Please, if you have a better idea, PR it, or let me know.

**Comment by jhheider** [comment]
Instead of using slugified versions of boards/lists/cards, I might well replace them with a ordinal identifier. So rather than:

cards in this list:
 - Test email                            > "Hey Kyle, does this work? -Sean  "
 - This is a sample requestion           > "I don’t need anything.  "
 - test2                                 > "test2  "
 - test                                  > "test  "
 - test                                  > "test  "
 - Printing Prices and Info              > "Hello, I am currently looking at getting…”

You could do this:

cards in this list:
 [1] Test email                            > "Hey Kyle, does this work? -Sean  "
 [2] This is a sample requestion           > "I don’t need anything.  "
 [3] test2                                 > "test2  "
 [4] test                                  > "test  "
 [5] test                                  > "test  "
 [6] Printing Prices and Info              > "Hello, I am currently looking at getting…”

Saves on auto-complete, and shouldn’t need to worry about duplicate names. I’ll see if I can whip something up one of these days.

**Comment by fiatjaf** [comment]
But that doesn't help when we want to quickly navigate to some card, @jhheider. That was the idea of the slugs, which allow us to type the first characters of the card and press tab for autocompletion.

**Comment by jhheider** [comment]
No, that's certainly true. Perhaps a configuration option or switch depending on preference.

**Comment by fiatjaf** [comment]
I don't believe there should be configuration options for this kind of behavior (or any kind of configuration option in the whole trelew app), nor that should be ordinal identifiers (at least not for now). What should be done here is to come up with a solution for the duplicated slugs, perhaps in the same way I tried and failed (but could have succeeded if I were a little more clever/patient), perhaps with a different solution.

Also, there is the problem that, when a _list_ has the same name of a _card_, their _vorpal aliases_ will coincide and a fatal error will occur, so something must be done. I believe my failed approach -- if succeeded -- would solve this, but I'm not sure.

codemill${5}

**Comment by fiatjaf** [comment]
I don't believe there should be configuration options for this kind of behavior (or any kind of configuration option in the whole trelew app), nor that should be ordinal identifiers (at least not for now). What should be done here is to come up with a solution for the duplicated slugs, perhaps in the same way I tried and failed (but could have succeeded if I were a little more clever/patient), perhaps with a different solution.

Also, there is the problem that, when a _list_ has the same name of a _card_, their _vorpal aliases_ will coincide and a fatal error will occur, so something must be done. I believe my failed approach -- if succeeded -- would solve this, but I'm not sure.

codemill$5

**Comment by palanglung** [comment]
I would suggest an array of command values so for example when i want to navigate to "board welcome" but there is more than 1 board named `welcome` it just simply ask which one with the list of the board with the same name and just answer it with the number from 1 to n.. for example

```bash
$ help

board welcome  enter board 'welcome' [2 board named welcome]

$ board welcome

1. welcome members 1 list 10
2. welcome members 1 list 5

which one: 1

welcome~$

```

**Comment by fiatjaf** [comment]
That's an awesome solution, although difficult to implement.
